iceoryx_posh/source/roudi/application/roudi_main.cpp

```diff
@@ -24,23 +25,28 @@ int main(int argc, char* argv[])
     using iox::roudi::IceOryxRouDiApp;
 
     iox::config::CmdLineParserConfigFileOption cmdLineParser;
-    cmdLineParser.parse(argc, argv);
-
-    iox::config::TomlRouDiConfigFileProvider configFileProvider(cmdLineParser);
-
-    iox::RouDiConfig_t roudiConfig =
-        configFileProvider.parse()
-            .or_else([](iox::roudi::RouDiConfigFileParseError& parseResult) {
-                iox::LogFatal() << "Couldn't parse config file. Error: "
-                                << iox::cxx::convertEnumToString(iox::roudi::ROUDI_CONFIG_FILE_PARSE_ERROR_STRINGS,
-                                                                 parseResult);
-                std::terminate();
-            })
-            .value();
-
-    cmdLineParser.printParameters();
-
-    IceOryxRouDiApp roudi(cmdLineParser, roudiConfig);
+    auto cmdLineArgs = cmdLineParser.parse(argc, argv);
+    if (cmdLineArgs.has_error())
```

Comment:
Again, do we need default behaviour in case of error ?

Reply:
See in previous comment. Topic for a follow-up pull request